### PR TITLE
Updates to multifactor authentication docs

### PR DESCRIPTION
**Pull request by erjohnson**
- [x] Remove beta frontmatter
- [x] Remove early access note
- [x] Use `MFA_API_ACCESS_TOKEN` in examples
- [x] Add links to sidebar
- [x] Rename / come up with better names for links in sidebar
- [x] Add token exchange example

https://auth0-docs-content-pr-6035.herokuapp.com/docs/multifactor-authentication/api

### CODE REVIEW

File: articles/multifactor-authentication/api/index.md [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
-The Multifactor Authentication (MFA) API is a set of APIs that allows you to:
-
-* Enforce MFA when users interact with [the Token endpoints](/api/authentication#get-token)
-* Programmatically enroll and manage user authenticators
+The Multifactor Authentication (MFA) API endpoints allow you to enforce MFA when users interact with [the Token endpoints](/api/authentication#get-token), as well programmatically enroll and manage user authenticators.
 
 ## Requirements
 
-To use the MFA APIs you will need **an enabled MFA rule that sets Guardian as the MFA provider**. Fore more info on this see [Guardian for Administrators](/multifactor-authentication/administrator).
+To use the MFA APIs, an enabled MFA rule that sets Guardian as the MFA provider is required. For more information, see [Guardian for Administrators](/multifactor-authentication/administrator).
```
Comment (cabralmartin) [inline articles/multifactor-authentication/api/index.md]: I missed one requirement. Clients need the `mfa` grant enabled. (This can be enabled in Applications -> Select app -> Advance Settings -> Client Grants -> Select MFA)

File: articles/multifactor-authentication/api/index.md [hunk 2]
```diff
@@ -1,28 +1,25 @@
...
 
 We have expanded MFA support on the Token endpoints to cover the following use cases:
 
-* Use MFA with the [password](/api-auth/grant/password), [password-realm](/api-auth/grant/password#realm-support), [refresh-token](/tokens/refresh-token/current#use-a-refresh-token) grants
-* Completion of first-time enrollment by end-users during authentication
-* Selection of the desired MFA authenticator by the end-user before they execute the MFA challenge
+* Use MFA with the [password](/api-auth/grant/password), [password-realm](/api-auth/grant/password#realm-support), and [refresh-token](/tokens/refresh-token/current#use-a-refresh-token) grants.
+* Completion of first-time enrollment by users during authentication.
+* Selection of the desired MFA authenticator by the user before they execute the MFA challenge.
 <!-- * Support for TOPT delivered via Email -->
```
Comment (cabralmartin) [inline articles/multifactor-authentication/api/index.md]: Are we tracking these pending TODOs in trello or similar? These are the next steps we need to execute.

File: articles/multifactor-authentication/api/_before-you-start.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+## Before you start
+
+The MFA endpoints require an [Access Token](/tokens/access-token) with:
+
+* `audience`: Set to `https://${account.namespace}/mfa`
```
Comment (cabralmartin) [inline articles/multifactor-authentication/api/_before-you-start.md]: Can we add a sample token?

File: articles/multifactor-authentication/api/oob.md [hunk 4]
```diff
@@ -1,14 +1,17 @@
...
-# Associate an OOB Authenticator
 
-In this tutorial, we will show you how you can configure your Auth0 tenant to allow the self-association of out-of-band (OOB) authenticators.
+# Associate an Out-of-Band Authenticator
 
-## Step 1. Trigger the MFA Error and Use the MFA Token to Associate the New Authenticator
+In this tutorial, you'll learn how to configure your application so users can self-associate out-of-band (OOB) authenticators.
 
-Whenever a user begins the authorization process and they do not have an active authenticator associated with their account, they will trigger the following MFA response when calling the `/oauth/token` endpoint:
+<%= include('./_before-you-start.md') %>
```
Comment (cabralmartin) [inline articles/multifactor-authentication/api/oob.md]: The current _before-you-start_ snippet is good for the Associate API, but doesn't work here. The tutorials that interact with the Token Endpoint should display the proper requirements which are similar to https://github.com/auth0/docs/pull/6035/files#diff-0a46c4ed6966526418bf5fc42c32d2b8R10. These requirements are more about the configuration required in the tenant.

File: articles/multifactor-authentication/api/otp.md [hunk 5]
```diff
@@ -1,14 +1,17 @@
...
-# Associate an OTP Authenticator
 
-In this tutorial, we will show you how you can configure your Auth0 tenant to allow the self-association of one-time password (OTP) authenticators.
+# Associate a One-Time Password Authenticator
 
-## Step 1. Trigger the MFA Error and Use the MFA Token to Associate the New Authenticator
+In this tutorial, you'll learn how to configure your application so users can self-associate one-time password (OTP) authenticators.
 
-Whenever a user begins the authorization process and they do not have an active authenticator associated with their account, they will trigger the following MFA response when calling the `/oauth/token` endpoint:
+<%= include('./_before-you-start.md') %>
```
Comment (cabralmartin) [inline articles/multifactor-authentication/api/otp.md]: Same as above

**Review by cabralmartin — CHANGES_REQUESTED** [review]

File: config/sidebar.yml [hunk 6]
```diff
@@ -549,6 +549,20 @@ articles:
             url: "/multifactor-authentication/developer/step-up-authentication/step-up-for-apis"
             hidden: true
 
+      - title: "API"
```
Comment (cabralmartin) [inline config/sidebar.yml:542]: Lets leave the addition in the sidebar for when the feature goes live for all customers. We should have the PR with the changeset ready, standing by for Product green light.

File: articles/multifactor-authentication/api/index.md [hunk 7]
```diff
@@ -1,69 +1,53 @@
...
+
+* Enable the MFA grant type for your application. You can enable the MFA grant by going to [Applications > Your Application > Advanced Settings > Grant Types](${manage_url}/#/applications) and selecting MFA.
+* Create a rule that sets Guardian as the MFA provider. For more information, see [Guardian for Administrators](/multifactor-authentication/administrator).
 
 If you are using the MFA API in conjunction with a [Token endpoint](/api/authentication#get-token), you must meet the requirements of the corresponding grant.
 
-## Enforcing MFA when interacting with the Token Endpoint
+## Multifactor authentication with the Token endpoint
 
 We have expanded MFA support on the Token endpoints to cover the following use cases:
```
Comment (mpaktiti) [inline articles/multifactor-authentication/api/index.md]: I really don't understand how this fits in in this doc. Looks completely out of context to me. Is this relevant to the API or MFA in general?

File: articles/multifactor-authentication/api/index.md [hunk 8]
```diff
@@ -1,69 +1,53 @@
...
-
-You can use this API to power your end-user interfaces where your users can:
-
-* Manage their MFA enrollments
-* Add new MFA authenticators
-* Remove existing MFA authenticators
+## Enrollment and management of user authenticators
 
-This is useful if you want your users to be able to:
+The MFA Associate API allows you to create, read, update, and delete authenticators. You can use this API to power user interfaces where users can manage MFA enrollments, or add and remove authenticators.
```
Comment (mpaktiti) [inline articles/multifactor-authentication/api/index.md]: sounds like this is a different API, while it's 3 different endpoints that are all part of the Authentication API (I would go as far as to ask why, since this feels like it should be part of the Mgmt API, but that's a product decision)

File: articles/multifactor-authentication/api/index.md [hunk 9]
```diff
@@ -1,69 +1,53 @@
...
-<!-- * [Enroll a new authenticator](/multifactor-authentication/api/manage#enroll-authenticators) -->
-<!-- TODO: Add link to enroll authenticator (requires expanding doc) -->
 :::
 
 ## Limitations
 
 * The MFA API is designed to work with the Guardian Provider. Support for other providers will be provided in future releases.
 
-* Support for authenticator selection is currently limited to the Token Endpoint. Auth0 is working to extend support to  Hosted MFA Pages. If users have more than one authenticator enrolled, the most-recently enrolled option will be used by the Hosted MFA Pages.
+* Support for authenticator selection is currently limited to the Token Endpoint. Auth0 is working to extend support to  Hosted MFA Pages. If users have more than one authenticator enrolled, the most-recently enrolled option will be used by the Hosted MFA Pages.
```
Comment (mpaktiti) [inline articles/multifactor-authentication/api/index.md]: link hosted MFA pages, people might not know what that is

File: articles/multifactor-authentication/api/index.md [hunk 10]
```diff
@@ -1,69 +1,53 @@
...
+The MFA Associate API allows you to create, read, update, and delete authenticators. You can use this API to power user interfaces where users can manage MFA enrollments, or add and remove authenticators.
 
-* Enroll more than one device
-* Select a fallback MFA mechanism in case the primary one is not available in a certain context. For example, your user might use OTP when their SMS network is not present or unresponsive
+This enables users to enroll more than one device and select a fallback MFA mechanism in case the primary one is not available. For example, your user might use OTP when their SMS network is not present or unresponsive.
 
 ### More info
 
 ::: next-steps
 * [List authenticators](/multifactor-authentication/api/manage#list-authenticators)
```
Comment (mpaktiti) [inline articles/multifactor-authentication/api/index.md]: That's the same page. Don't add two different links for the same page. Just call it "Manage the Authenticators"

Comment (mpaktiti) [reply articles/multifactor-authentication/api/index.md]: Personally I'm not fond of having next-steps element in the middle of the doc either. This is not a practice we follow, you won't find another doc that follows this.

File: articles/multifactor-authentication/api/manage.md [hunk 11]
```diff
@@ -1,16 +1,34 @@
...
-This article includes documentation on features that are still under development. These features are available to customers with early access.
-:::
+Auth0 provides several API endpoints to help you manage the authenticators you're using with an application for multifactor authentication (MFA).
 
-Auth0 provides several API endpoints to help you manage the authenticators you're using with a particular tenant for multifactor authentication (MFA).
+## Before you start
+
+The MFA endpoints require an [Access Token](/tokens/access-token) with:
+
+- `audience`: Set to `https://${account.namespace}/mfa`
```
Comment (mpaktiti) [inline articles/multifactor-authentication/api/manage.md:14]: what if I have a custom domain? Is this value affected? should I use my custom domain instead?

File: articles/multifactor-authentication/api/manage.md [hunk 12]
```diff
@@ -1,16 +1,34 @@
...
+- `scope`: Include `enroll` for enrollment, `read:authenticators` to list authenticators, and `remove:authenticators` to delete authenticators.
```
Comment (mpaktiti) [inline articles/multifactor-authentication/api/manage.md:15]: This is **really important information** and I don't remember reading it in the other PR, for the Authentication API. People must know how to get a valid token for each endpoint.

File: articles/multifactor-authentication/api/manage.md [hunk 13]
```diff
@@ -22,7 +40,7 @@ To get a list of the authenticators you've associated and can be used with your
 	"url": "https://${account.namespace}/mfa/authenticators",
 	"headers": [{
 		"name": "Authorization",
-		"value": "Bearer YOUR_API_ACCESS_TOKEN"
+		"value": "Bearer MFA_API_ACCESS_TOKEN"
```
Comment (mpaktiti) [inline articles/multifactor-authentication/api/manage.md]: This looks like a good old-fashioned, access token and I would strongly recommend naming this what it is: YOUR_ACCESS_TOKEN

We use terms like "Management APIv2 Token" and "MFA API Token" and we make people think that this is some other type of token, while it's **not**. Same old access token, only the scopes change. Follow that please also in the Authentication API PR.

Ignore my comment if this is not an access token but sth else.

File: articles/multifactor-authentication/api/oob.md [hunk 14]
```diff
@@ -1,14 +1,22 @@
...
 
-In this tutorial, we will show you how you can configure your Auth0 tenant to allow the self-association of out-of-band (OOB) authenticators.
+# Associate an Out-of-Band Authenticator
 
-## Step 1. Trigger the MFA Error and Use the MFA Token to Associate the New Authenticator
+In this tutorial, you'll learn how to configure your application so users can self-associate out-of-band (OOB) authenticators.
 
-Whenever a user begins the authorization process and they do not have an active authenticator associated with their account, they will trigger the following MFA response when calling the `/oauth/token` endpoint:
+## Before you start
+
```
Comment (mpaktiti) [inline articles/multifactor-authentication/api/oob.md]: reuse content everywhere you can. I saw the same section at the articles/multifactor-authentication/api.md doc

File: articles/multifactor-authentication/api/otp.md [hunk 15]
```diff
@@ -1,14 +1,22 @@
...
+# Associate a One-Time Password Authenticator
 
-## Step 1. Trigger the MFA Error and Use the MFA Token to Associate the New Authenticator
+In this tutorial, you'll learn how to configure your application so users can self-associate one-time password (OTP) authenticators.
 
-Whenever a user begins the authorization process and they do not have an active authenticator associated with their account, they will trigger the following MFA response when calling the `/oauth/token` endpoint:
+## Before you start
+
+Before you can use the MFA APIs, you'll need to:
+
```
Comment (mpaktiti) [inline articles/multifactor-authentication/api/otp.md]: reuse this section

File: articles/multifactor-authentication/api/otp.md [hunk 16]
```diff
@@ -1,14 +1,22 @@
...
+* Enable the MFA grant type for your application. You can enable the MFA grant by going to [Applications > Your Application > Advanced Settings > Grant Types](${manage_url}/#/applications) and selecting MFA.
+* Create a rule that sets Guardian as the MFA provider. For more information, see [Guardian for Administrators](/multifactor-authentication/administrator).
+
+## 1. Get the MFA Token
+
```
Comment (mpaktiti) [inline articles/multifactor-authentication/api/otp.md]: shouldn't you have the request also here? you only have the response.

**Review by mpaktiti — CHANGES_REQUESTED** [review]

File: articles/multifactor-authentication/api/manage.md (hunk 11 above)
Comment (erjohnson) [reply articles/multifactor-authentication/api/manage.md:14]: @cabralmartin

**Review by mpaktiti — APPROVED** [review]
LGTM. I'll merge alongside https://github.com/auth0/docs/pull/6008 once the last changes are applied.

File: articles/multifactor-authentication/api/manage.md (hunk 11 above)
Comment (cabralmartin) [reply articles/multifactor-authentication/api/manage.md:14]: Value not affected.
